### PR TITLE
thread search integration

### PR DESCRIPTION
**Pull request by mmaxim**
Hooks up thread search into the UI.

### CODE REVIEW

**Comment by mmaxim** [comment]
Requires https://github.com/keybase/client/pull/16601

File: shared/chat/conversation/search/index.desktop.js [hunk 1]
```diff
@@ -3,69 +3,181 @@
...
-                  type="iconfont-arrow-down"
-                />
-              </Kb.Box2>
-            )}
+      <Kb.Box2 direction="vertical" fullWidth={true} style={this.props.style}>
+        <Kb.Box2 direction="horizontal" style={styles.outerContainer} fullWidth={true} gap="tiny">
+          <Kb.Box2 direction="horizontal" style={styles.inputContainer} fullWidth={true}>
+            <Kb.Box2 direction="horizontal" gap="xtiny" fullWidth={true} centerChildren={true}>
+              <Kb.Icon type="iconfont-search" color={Styles.globalColors.black_50} fontSize={16} />
+              <Kb.PlainInput
```
Comment (buoyad) [inline shared/chat/conversation/search/index.desktop.js]: when switching between convos you're searching in, the results stay consistent with the conversation but the query doesn't.

File: shared/chat/conversation/search/index.desktop.js [hunk 2]
```diff
@@ -3,69 +3,181 @@
...
+                onBlur={this._onBlur}
+                onChangeText={this._onChangedText}
+                onEnterKeyDown={this._submitSearch}
+                onFocus={this._onFocus}
+                onKeyDown={this._onKeydown}
+                placeholder="Search..."
+              />
+            </Kb.Box2>
+            <Kb.Box2 direction="horizontal" gap="tiny" style={styles.resultsContainer}>
+              {this.props.inProgress && <Kb.ProgressIndicator style={styles.progress} />}
```
Comment (buoyad) [inline shared/chat/conversation/search/index.desktop.js]: Should have some way to communicate there were no hits

File: shared/chat/conversation/search/index.desktop.js [hunk 3]
```diff
@@ -3,69 +3,181 @@
...
           </Kb.Box2>
+          <Kb.Button
+            type="Primary"
+            disabled={this.props.inProgress}
+            onClick={this._submitSearch}
+            label="Search"
+          />
+          <Kb.Button
+            type="Secondary"
+            disabled={!this.props.inProgress}
```
Comment (buoyad) [inline shared/chat/conversation/search/index.desktop.js]: It seems weird that this is ever disabled

File: shared/chat/conversation/search/container.js [hunk 4]
```diff
@@ -1 +1,71 @@
...
+}
+
+const mapDispatchToProps = (dispatch, {conversationIDKey}) => ({
+  _loadSearchHit: messageID =>
+    dispatch(Chat2Gen.createLoadMessagesFromSearchHit({conversationIDKey, messageID})),
+  onCancel: () => dispatch(Chat2Gen.createCancelThreadSearch()),
+  onHotkey: (cmd: string) => {
+    switch (cmd) {
+      case 'esc':
+        dispatch(Chat2Gen.createToggleThreadSearch({conversationIDKey}))
```
Comment (buoyad) [inline shared/chat/conversation/search/container.js]: This should cancel right?

File: shared/reducers/chat2.js [hunk 5]
```diff
@@ -1007,6 +1007,27 @@ const rootReducer = (
       return state.update('unsentTextMap', old =>
         old.setIn([action.payload.conversationIDKey], action.payload.text)
       )
+    case Chat2Gen.threadSearchResult:
+      return state.updateIn(['threadSearchInfoMap', action.payload.conversationIDKey], info =>
+        info.set('hits', info.hits.push(action.payload.message))
+      )
+    case Chat2Gen.setThreadSearchInProgress:
+      return state.updateIn(['threadSearchInfoMap', action.payload.conversationIDKey], info => {
+        return (info || Constants.makeThreadSearchInfo()).set('inProgress', action.payload.inProgress)
```
Comment (buoyad) [inline shared/reducers/chat2.js]: :sheep: you could use default fcn args here and below to clean these a bit

File: shared/chat/conversation/search/container.js (hunk 4 above)
Comment (mmaxim) [reply shared/chat/conversation/search/container.js]: It does.

File: shared/chat/conversation/normal/container.js [hunk 6]
```diff
@@ -81,5 +100,8 @@ export default compose(
       }),
       onRequestScrollUp: ({scrollListUpCounter}) => () => ({scrollListUpCounter: scrollListUpCounter + 1}),
     }
-  )
-)(Normal)
+  ),
+  withProps(props => ({
```
[the remark this answers is not in the document]
Comment (mmaxim) [reply shared/chat/conversation/normal/container.js]: No

File: shared/chat/conversation/header-area/normal/container.js [hunk 7]
```diff
@@ -38,13 +39,14 @@ const mapStateToProps = (state, {infoPanelOpen, conversationIDKey, isPending}) =
...
 
-const mapDispatchToProps = (dispatch, {onToggleInfoPanel, conversationIDKey}) => ({
+const mapDispatchToProps = (dispatch, {onToggleInfoPanel, onToggleThreadSearch, conversationIDKey}) => ({
   _onCancel: () => dispatch(Chat2Gen.createSetPendingMode({pendingMode: 'none'})),
   _onOpenFolder: () => dispatch(Chat2Gen.createOpenFolder({conversationIDKey})),
   _onUnMuteConversation: () => dispatch(Chat2Gen.createMuteConversation({conversationIDKey, muted: false})),
   onBack: () => dispatch(RouteTreeGen.createNavigateUp()),
   onShowProfile: (username: string) => dispatch(createShowUserProfile({username})),
   onToggleInfoPanel,
+  onToggleThreadSearch,
```
[the remark this answers is not in the document]
Comment (mmaxim) [reply shared/chat/conversation/header-area/normal/container.js]: Yes

**Comment by mmaxim** [comment]
@buoyad PTAL

**Pull request merged**